llvm/lib/Transforms/InstCombine/InstCombineCasts.cpp

```diff
@@ -122,6 +125,19 @@ static Value *EvaluateInDifferentTypeImpl(Value *V, Type *Ty, bool isSigned,
         Res = CallInst::Create(Fn->getFunctionType(), Fn);
         break;
       }
+      case Intrinsic::umin:
+      case Intrinsic::umax: {
+        Value *LHS = IC.InsertNewInstWith(
+            CastInst::CreateIntegerCast(I->getOperand(0), Ty, isSigned),
+            I->getIterator());
```

Comment:
This should be calling EvaluateInDifferentTypeImpl instead of manually inserting a cast.